packages/at_lookup/lib/src/at_lookup_impl.dart

```diff
@@ -628,19 +642,33 @@ class AtLookupImpl implements AtLookUp {
         !(_connection!.getMetaData()!.isAuthenticated);
   }
 
-  Future<bool> createOutBoundConnection(String host, String port,
-      String toAtSign, SecureSocketConfig secureSocketConfig) async {
+  Future<bool> createOutboundConnection(
+      String host, String port, SecureSocketConfig secureSocketConfig) async {
     try {
-      SecureSocket secureSocket =
-          await socketFactory.createSocket(host, port, secureSocketConfig);
-      _connection =
-          outboundConnectionFactory.createOutboundConnection(secureSocket);
+      // Create the socket connection using the factory
+      final underlying = await atOutboundConnectionFactory.createUnderlying(
+          host, port, secureSocketConfig);
+
+      // Create the outbound connection and listener using the factory's methods
+      final outboundConnection =
+          atOutboundConnectionFactory.outBoundConnectionFactory(underlying);
+      messageListener = atOutboundConnectionFactory
+          .atLookupSocketListenerFactory(outboundConnection);
+
+      // Set the connection type in `_webSocketConnection` or `_connection`
+      if (underlying is WebSocket) {
```

Comment:
this type casting has no impact on the code since we are not accessing any attributes specific to OutboundWebSocketConnection or OutboundConnection
If _connection is not set before, we can just have 
_connection = outboundConnection

Reply:
Updated